__tests__/commands/run.js

```diff
@@ -0,0 +1,31 @@
+/* @flow */
+
+import {run} from '../../src/cli/commands/run.js';
+import * as reporters from '../../src/reporters/index.js';
+import Config from '../../src/config.js';
+
+jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;
+
+const path = require('path');
+
+const fixturesLoc = path.join(__dirname, '..', 'fixtures', 'run');
+
+async function runRun(
+  flags: Object,
+  args: Array<string>,
+  name: string,
+): Promise<void> {
+  const cwd = path.join(fixturesLoc, name);
+  const reporter = new reporters.NoopReporter();
+  const config = new Config(reporter);
+  await config.init({cwd});
+  return run(config, reporter, flags, args);
+}
+
+test.concurrent('run should run script', (): Promise<void> => {
+  return runRun({}, ['some-script'], 'run-should-run-script');
+});
+
+test.concurrent('run should run binary', (): Promise<void> => {
+  return runRun({}, ['echo'], 'run-should-run-binary');
+});
```

Comment:
Come to think of it, it could be reasonable to mock out `child_process` (used _all the way_ [over here](https://github.com/yarnpkg/yarn/blob/0f1db96846badbbe3b43c470c05196eb29437fb4/src/util/child.js#L9)) for these purposes, but I guess it depends if you want to actually integration test with the shell.